internal/rpc/verifydeposit.go

```diff
@@ -25,12 +24,13 @@ func (s *UtxoServer) VerifyDeposit(tx wire.MsgTx, evmAddress string) (isTrue boo
 		network = &chaincfg.TestNet3Params
 	}
 
-	pubKey, err := s.getPubKey()
+	// invalid version: 100
+	version := uint32(100)
+	pubKey, err := s.state.GetPubKey()
 	if err != nil {
-		return false, 100, err
+		return false, 100, "", err
 	}
 
-	version := uint32(100)
 	for _, out := range tx.TxOut {
 		pkScript := hex.EncodeToString(out.PkScript)
```

Comment:
Please use IsUtxoGoatDepositV1 directly if evmAddress is empy, else use IsUtxoGoatDepositV0
Many codes are not necerssary.